src/main/java/com/bookshop/bazydanych/product/ProductService.java

```diff
@@ -75,10 +83,49 @@ public Product getbyId(Long id){
         return productRepository.getById(id);
     }
 
-    public void updateProduct(Product product){
-        Product productToUpdate = productRepository.getOne(product.getId());
-        productToUpdate = product;
-        productRepository.save(productToUpdate);
-        //TODO dodawanie produktów!
+    public void updateProduct(Product productOld){
+        Product product = new Product(
+                productOld.getName(),
+                productOld.getUnit(),
+                productOld.getProducent(),
+                productOld.getStock(),
+                productOld.getPrice(),
+                productOld.getStatus(),
+                productOld.getDescription(),
+                productOld.getCurrency(),
+                productOld.getCategory(),
+                productOld.getPlatform());
+
+        ArrayList<Long> orders = new ArrayList<>();
+        orderService.getAllOrders().forEach(a -> {
+            a.getProductIds().forEach(b ->{
+                if(b == product.getId()){
+                    if(orders.indexOf(a.getId())<0){
+                        orders.add(a.getId());
+                    }
+                }
+            });
+        });
+
+        if(orders.isEmpty()){
+            getActive().forEach(a -> {
+                if(a.getName() == product.getName()){
+                    deactivateProduct(a.getId());
+                }
+            });
+        } else {
+            getActive().forEach(a -> {
+                if(a.getName() == product.getName()){
+                    deactivateProduct(a.getId());
+                }
+            });
+            productReservationRepository.findAll().forEach(a -> {
+                if(a.getProductId() == productRepository.getByName(product.getName()).getId()){
```

Comment:
damn stary za każdym razem pobierasz ten sam obiekt z bazy, ten foreach będzie wolniejszy niż pętla w pythonie, tak się nie robi xD nie wiem czemu nie użyjesz bazy danych tylko wszędzie wyżej jakoś dziwnie filtrujesz to wszystko javowymi funkcjami jak możesz połowe operacji na bazie wykonać

Reply:
Nie wiem jak to na bazie zrobić :/